sw/host/opentitanlib/src/bootstrap/rescue.rs

```diff
@@ -209,10 +209,10 @@ impl Rescue {
 
     /// Reads and discards any characters in the receive buffer, waiting a little while for any
     /// more which will also be discarded.
-    fn flush_rx(&self, uart: &dyn Uart) {
+    fn flush_rx(&self, uart: &dyn Uart, timeout: Duration) {
         let mut response = [0u8; Frame::HASH_LEN];
         loop {
-            match uart.read_timeout(&mut response, Duration::from_millis(500)) {
+            match uart.read_timeout(&mut response, timeout) {
```

Comment:
Do you want `timeout` to decrement with each try? (ie: is the parameter the "total" timeout for `flush_rx` or is it the amount to wait each time in the loop?)

Reply:
I do not believe the distinction is important.  The code is meant to ensure a buffer is emptied, so in practice one or two initial iterations would return data immediately, and then the last iteration would wait the full timeout.

I made the delay configurable, because then I could conveniently use the buffer-emptying routine to guarantee the minimum width of the reset pulse.